cli/command/service/logs.go

```diff
@@ -66,26 +73,91 @@ func runLogs(dockerCli *command.DockerCli, opts *logsOptions) error {
 	}
 
 	client := dockerCli.Client()
+
+	service, _, err := client.ServiceInspectWithRaw(ctx, opts.service)
+	if err != nil {
+		return err
+	}
+
 	responseBody, err := client.ServiceLogs(ctx, opts.service, options)
 	if err != nil {
 		return err
 	}
 	defer responseBody.Close()
 
-	resolver := idresolver.New(client, opts.noResolve)
+	var replicas uint64
```

Comment:
Hmm, rather than rely on this we could move the `padding = len(strconv.FormatUint(replicas, 10))` inside the `if` and initialize it to a default value of 1? That will make it harder to unintentionally use `replicas` if it hasn't been initialized.